examples/batch_dispatching.rs

```diff
@@ -0,0 +1,149 @@
+//! This example show how to use the Batch dispatcher.
+//!
+//! The Batch is a feature that allow to control the dispatching of a group of
+//! systems.
+//!
+//! Specifically here we have three Systems
+//! - **SayHelloSystem**: Which is directly registered under the main
+//!   dispatcher.
+//! - **BuyTomatoSystem** and **BuyPotatoSystem** are registered in the batch.
+//!
+//! Notice that none of them is using any direct dependency.
+//! The SayHelloSystem is requesting the resources **TomatoStore** and
+//! **PotatoStore**, which are also requested by the other two systems inside
+//! the batch and by the batch controller itself.
+//!
+//! This to demonstrate that the dispatcher is able to correctly organize, their
+//! execution.
+//!
+//! Is also showed how to use a custom BatchControllerSystem.
+//! Check the **CustomBatchControllerSystem** which execute its inner Systems
+//! three times.
+
+use shred::{
+    AccessorCow, BatchAccessor, BatchController, BatchUncheckedWorld, Dispatcher,
+    DispatcherBuilder, Read, RunningTime, System, World, Write,
+};
+use std::{thread::sleep, time::Duration};
+
+fn main() {
+    let mut dispatcher = DispatcherBuilder::new()
+        .with(SayHelloSystem, "say_hello_system", &[])
+        .with_batch::<CustomBatchControllerSystem>(
+            DispatcherBuilder::new()
+                .with(BuyTomatoSystem, "buy_tomato_system", &[])
+                .with(BuyPotatoSystem, "buy_potato_system", &[]),
+            "BatchSystemTest",
+            &[],
+        )
+        .build();
+
+    let mut world = World::empty();
+
+    dispatcher.setup(&mut world);
+
+    // Running phase
+    for i in 0..10 {
+        println!("Dispatching {} ", i);
+
+        dispatcher.dispatch(&world);
+        sleep(Duration::new(0, 100000000));
+    }
+
+    // Done
+    println!("Execution finished");
+}
+
+// Resources
+
+#[derive(Default)]
+pub struct PotatoStore(i32);
+
+#[derive(Default)]
+pub struct TomatoStore(f32);
+
+// Say Hello Sytem
+
+pub struct SayHelloSystem;
+
+impl<'a> System<'a> for SayHelloSystem {
+    type SystemData = (Write<'a, PotatoStore>, Write<'a, TomatoStore>);
+
+    fn run(&mut self, _data: Self::SystemData) {
+        println!("Hello!")
+    }
+}
+
+// BuyPotatoSystem
+
+pub struct BuyPotatoSystem;
+
+impl<'a> System<'a> for BuyPotatoSystem {
+    type SystemData = (Write<'a, PotatoStore>);
+
+    fn run(&mut self, _data: Self::SystemData) {
+        println!("Buy Potato")
+    }
+}
+
+// Buy Tomato System
+
+pub struct BuyTomatoSystem;
+
+impl<'a> System<'a> for BuyTomatoSystem {
+    type SystemData = (Write<'a, PotatoStore>);
```

Comment:
Is this supposed to be `PotatoStore`?

Reply:
In theory yes, to show that is dispatched in sequence due to the dependency.
I'll change it since doesn't make sense and I already have a sequential dispatching in the tests.